README.md

```diff
@@ -42,17 +42,32 @@ Approvals.verify(your_subject, :format => :json)
 
 This will raise an `ApprovalError` in the case of a failure.
 
+The first time the approval is run, a file will be created with the contents of the subject of your approval:
+
+    the_name_of_the_approval.received.txt # or .json, .html, .xml as appropriate
+
+Since you have not yet approved anything, the `*.approved` file does not exist, and the comparison will fail.
+
+### Customizing formatted output
+
 The default writer uses the `:to_s` method on the subject will be used to generate the output for
-the `received` file. For custom complex objects you will need to override
-`:to_s` to get helpful output, rather than the default:
+the `received` file. For custom complex objects you will need to provide a custom writer to get helpful output, rather than the default:
 
     #<Object:0x0000010105ea40> # or whatever the object id is
 
-The first time the approval is run, a file will be created with the contents of the subject of your approval:
+Create a `CustomWriter` class somewhere accessible to your test:
 
-    the_name_of_the_approval.received.txt # or .json, .html, .xml as appropriate
+```
+class CustomWriter < TextWriter
+  def format(data)
+    # Custom data formatting here
+  end
 
-Since you have not yet approved anything, the `*.approved` file does not exist, and the comparison will fail.
+  def filter(data)
+    # Custom data filtering here
+  end
+end
+```
```

Comment:
We should also probably document that when you want to use the `CustomWriter` you'll need to use `verify(object, format: :custom)` or some such, otherwise the default remains `TextWriter`